File: elixir/day3/lib/part1.ex
defmodule Part1 do
  def extract_and_sum_multiplications(input) do
    ~r/mul\((\d{1,3}),(\d{1,3})\)/
    |> Regex.scan(input, capture: :all_but_first)
    |> Enum.map(fn [x, y] ->
      String.to_integer(x) * String.to_integer(y)
    end)
    |> Enum.sum()
  end

  def part1(input) do
    extract_and_sum_multiplications(input)
  end
end

File.read!(~c"input.txt")
|> Part1.part1()
|> IO.inspect(label: "part1")

File: ruby/day3/part1.rb
# frozen_string_literal: true

def part1(input)
  input.scan(/mul\((\d+),(\d+)\)/).sum { |a, b| a.to_i * b.to_i }
end

if __FILE__ == $PROGRAM_NAME
  input = File.read('input.txt')
  p part1(input)
end
